File: implementations/react/src/patterns/structural/facade/subsystems/calendar-subsystem.ts
import type { Slot } from "../types/types";

/**
 * Subsistema de calendario que maneja reservas y cancelaciones de citas.
 */

export class CalendarAPI {
  async reserve(slot: Slot) {
    return { reservationId: "res_" + Math.random().toString(36).slice(2), slot };
  }
  async cancel(reservationId: string) {
    return { cancelled: true, reservationId };
  }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/structural/facade/subsystems/calendar_subsystem.dart
class CalendarReservation {
  final String reservationId;
  CalendarReservation(this.reservationId);
}

class CalendarAPI {
  Future<CalendarReservation> reserve(Map<String, dynamic> slot) async {
    await Future.delayed(const Duration(milliseconds: 100));
    return CalendarReservation('res-${DateTime.now().millisecondsSinceEpoch}');
  }

  Future<void> cancel(String reservationId) async {
    await Future.delayed(const Duration(milliseconds: 60));
    return;
  }
}
